3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,58 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+const body = document.querySelector('body');
+const button = document.createElement('button');
+const select = document.createElement('select');
+const image = document.createElement('img');
+button.type = 'submit';
+image.style.width = '200px';
+button.textContent = 'Get Pokemon';
+body.appendChild(button);
+body.appendChild(select);
+body.appendChild(image);
+
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error('Network Error');
+  }
+  const data = await response.json();
+  return data.results;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+function fetchAndPopulatePokemons(data) {
   // TODO complete this function
+  data.forEach((item) => {
+    const option = document.createElement('option');
+    select.appendChild(option);
+    option.value = item.name;
+    option.textContent = item.name;
+  });
+  select.addEventListener('change', fetchImage);
 }
-
-function fetchImage(/* TODO parameter(s) go here */) {
+async function fetchImage() {
   // TODO complete this function
+  const response = await fetch(
+    `https://pokeapi.co/api/v2/pokemon/${select.value}`
+  );
+  const data = await response.json();
+  button.addEventListener('click', () => {
```

Comment:
Interesting solution. The original idea was to show the pokemon image every time I change the selection of the pokemon without clicking the button.

But I'll accept this solution as well

Reply:
Thank you Stasel